src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -986,8 +986,7 @@ CompressionCodecPtr IMergeTreeDataPart::detectDefaultCompressionCodec() const
     for (const auto & part_column : columns)
     {
         /// It was compressed with default codec and it's not empty
-        auto column_size = getColumnSize(part_column.name);
-        if (column_size.data_compressed != 0 && !storage_columns.hasCompressionCodec(part_column.name))
+        if (storage_columns.hasCompressionCodec(part_column.name))
```

Comment:
This was from some debugging I did and is unrelated to my change. I noticed that the previous if statement was never true and so the files never get read from disk in `detectDefaultCompressionCodec`